src/renderers/WebGLRenderer.js

```diff
@@ -719,31 +726,57 @@ function WebGLRenderer( parameters = {} ) {
 		//
 
 		let index = geometry.index;
-		const position = geometry.attributes.position;
+		let rangeFactor = 1;
+
+		if ( material.wireframe === true ) {
+
+			index = geometries.getWireframeAttribute( geometry );
+			rangeFactor = 2;
+
+		}
 
 		//
 
-		if ( index === null ) {
+		const range = { start: 0, end: Infinity };
 
-			if ( position === undefined || position.count === 0 ) return;
+		const drawRange = geometry.drawRange;
+		const position = geometry.attributes.position;
 
-		} else if ( index.count === 0 ) {
+		if ( drawRange !== null ) {
 
-			return;
+			clampRange( range, {
+				start: drawRange.start * rangeFactor,
+				end: ( drawRange.start + drawRange.count ) * rangeFactor,
+			} );
 
 		}
 
-		//
+		if ( group !== null ) {
 
-		let rangeFactor = 1;
+			clampRange( range, {
```

Comment:
Please avoid the creation of objects in functions/methods which are invocated per frame multiple times. Object creation (even inline) is an expensive operation and should not be done over and over again.

Reply:
Fair point. I put my faith in modern JavaScript engines to optimize that sort of thing, but it's probably better not to rely on it. I will update the code.